ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://tanzanite.cluster.engflow.com/invocations/default/101b275d-a0fc-4ec5-bd7f-64060184dad4?testReportRun=3&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [13866fc3eddb4f308f3f0bb56e513c7b32293da0](https://github.com/cockroachdb/cockroach/commits/13866fc3eddb4f308f3f0bb56e513c7b32293da0):


```
        	pkg/ccl/changefeedccl/main_test.go:28 +0x1bb
        main.main()
        	bazel-out/k8-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:720 +0x754
        goroutine 50 [select]:
        go.opencensus.io/stats/view.(*worker).start(0xc00105cc80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x1a5
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0xfd
        goroutine 23 [syscall, 6 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x29
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1d
        created by os/signal.Notify.func1.1 in goroutine 7
        	GOROOT/src/os/signal/signal.go:151 +0x47
        goroutine 6 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x6e
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x25
        goroutine 7 [chan receive, 6 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0xae
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x31
        goroutine 8 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x130
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x25
        goroutine 194056 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0040780e0, {0xcf7de00, 0xc004aeed70})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x1f1
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 194052
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x245
        goroutine 194055 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc004078060, {0xcf7de00, 0xc004aeed20})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x1f1
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 194052
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x245: Leaked goroutine: goroutine 194055 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc004078060, {0xcf7de00, 0xc004aeed20})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x1f1
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 194052
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x245
        Leaked goroutine: goroutine 194056 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0040780e0, {0xcf7de00, 0xc004aeed70})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x1f1
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 194052
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x245
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (240.53s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19464672?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19464672?buildTab=artifacts#/) on master @ [09a462e894e60c149627b998fd0530ed9b907c58](https://github.com/cockroachdb/cockroach/commits/09a462e894e60c149627b998fd0530ed9b907c58):


```
        	pkg/ccl/changefeedccl/main_test.go:28 +0x134
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:720 +0x398
        goroutine 5 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x40011dc200)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 36 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 37 [chan receive, 1 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 24 [syscall, 1 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 37
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 38 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 99164 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4009e6bf60, {0x7d1ac58, 0x4005799a90})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 99160
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 99163 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4009e6bf00, {0x7d1ac58, 0x4005799a40})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 99160
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: Leaked goroutine: goroutine 99163 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4009e6bf00, {0x7d1ac58, 0x4005799a40})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 99160
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 99164 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4009e6bf60, {0x7d1ac58, 0x4005799a90})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 99160
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (29.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot T-cdc branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19500539?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19500539?buildTab=artifacts#/) on master @ [0868f32a1e48ac0508699989aba0cf9ba7c0353e](https://github.com/cockroachdb/cockroach/commits/0868f32a1e48ac0508699989aba0cf9ba7c0353e):


```
        	pkg/ccl/changefeedccl/main_test.go:28 +0x134
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:720 +0x398
        goroutine 53 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x400101ac80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 55 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 56 [chan receive, 1 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 99 [syscall, 1 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 56
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 57 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 128357 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40017a6b60, {0x7d1cd58, 0x4008607b80})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 128321
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 128356 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40017a6800, {0x7d1cd58, 0x4008607b30})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 128321
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: Leaked goroutine: goroutine 128356 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40017a6800, {0x7d1cd58, 0x4008607b30})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 128321
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 128357 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40017a6b60, {0x7d1cd58, 0x4008607b80})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 128321
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (49.71s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19605140?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/19605140?buildTab=artifacts#/) on master @ [2dce1ce9c244b7e55485e50515b75325a16eb7d0](https://github.com/cockroachdb/cockroach/commits/2dce1ce9c244b7e55485e50515b75325a16eb7d0):


```
        	pkg/ccl/changefeedccl/main_test.go:28 +0x134
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:720 +0x398
        goroutine 50 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x4001001080)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 53 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 54 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 67 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 54
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 55 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 144956 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40029f58c0, {0x7dc3d18, 0x4007985130})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144952
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 144955 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40029f5860, {0x7dc3d18, 0x4007984ff0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144952
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: Leaked goroutine: goroutine 144955 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40029f5860, {0x7dc3d18, 0x4007984ff0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144952
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 144956 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40029f58c0, {0x7dc3d18, 0x4007985130})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144952
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (38.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20406677?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20406677?buildTab=artifacts#/) on master @ [4fe0aa8f7d17c09142cdde77b4a8b3ba3f9a95cd](https://github.com/cockroachdb/cockroach/commits/4fe0aa8f7d17c09142cdde77b4a8b3ba3f9a95cd):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:770 +0x398
        goroutine 13 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x40011fa200)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 53 [syscall, 3 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 66
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 16 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 66 [chan receive, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 67 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 189647 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40123c29a0, {0x7f92758, 0x40073c9bd0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 189644
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 189648 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40123c2a20, {0x7f92758, 0x40073c9c20})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 189644
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: 
        Leaked goroutine: goroutine 189647 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40123c29a0, {0x7f92758, 0x40073c9bd0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 189644
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 189648 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40123c2a20, {0x7f92758, 0x40073c9c20})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 189644
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (57.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20431832?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20431832?buildTab=artifacts#/) on master @ [e06b5d119f914c650c7341fda3fab6ee670dab1c](https://github.com/cockroachdb/cockroach/commits/e06b5d119f914c650c7341fda3fab6ee670dab1c):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:774 +0x398
        goroutine 20 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x400103fd00)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 23 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 7
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 6 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 7 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 8 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 161132 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40079374e0, {0x7fb9cd8, 0x400ac2a8c0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 161128
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 161131 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007937460, {0x7fb9cd8, 0x400ac2a870})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 161128
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: 
        Leaked goroutine: goroutine 161131 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007937460, {0x7fb9cd8, 0x400ac2a870})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 161128
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 161132 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40079374e0, {0x7fb9cd8, 0x400ac2a8c0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 161128
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (76.20s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-2 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20489151?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20489151?buildTab=artifacts#/) on master @ [7a39b97b473f507171f25b24190a8d2c8db72edf](https://github.com/cockroachdb/cockroach/commits/7a39b97b473f507171f25b24190a8d2c8db72edf):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:782 +0x398
        goroutine 54 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x400121c200)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 67 [syscall, 1 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 20
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 19 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 20 [chan receive, 1 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 21 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 144686 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4008808040, {0x800e138, 0x400c39d7c0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144683
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 144687 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40088080a0, {0x800e138, 0x400c39d810})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144683
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: 
        Leaked goroutine: goroutine 144686 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4008808040, {0x800e138, 0x400c39d7c0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144683
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 144687 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40088080a0, {0x800e138, 0x400c39d810})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 144683
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (67.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20650202?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20650202?buildTab=artifacts#/) on master @ [629623abf713f52f7fb5b6a801d875c9566df896](https://github.com/cockroachdb/cockroach/commits/629623abf713f52f7fb5b6a801d875c9566df896):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:792 +0x398
        goroutine 6 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x4001057b80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x88
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x98
        goroutine 38 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 39 [chan receive, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x74
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 82 [syscall, 3 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 39
        	GOROOT/src/os/signal/signal.go:151 +0x28
        goroutine 40 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x98
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 106892 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400a4021a0, {0x80a3198, 0x4001b00960})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 106888
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        goroutine 106891 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400a402140, {0x80a3198, 0x4001b00910})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 106888
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118: 
        Leaked goroutine: goroutine 106891 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400a402140, {0x80a3198, 0x4001b00910})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 106888
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
        Leaked goroutine: goroutine 106892 [select]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400a4021a0, {0x80a3198, 0x4001b00960})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 106888
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (67.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20713396?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20713396?buildTab=artifacts#/) on master @ [ddeef368172658dddb47811fe0fcdcaf5abbe8ee](https://github.com/cockroachdb/cockroach/commits/ddeef368172658dddb47811fe0fcdcaf5abbe8ee):


```
        main.main()
        	bazel-out/s390x-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:804 +0x5b4
        goroutine 43 [select]:
        go.opencensus.io/stats/view.(*worker).start(0xc0011e6b80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0xba
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0xd0
        goroutine 9 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x82
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x30
        goroutine 10 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0xa8
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x42
        goroutine 11 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0xe0
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x30
        goroutine 24 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x3e
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x24
        created by os/signal.Notify.func1.1 in goroutine 10
        	GOROOT/src/os/signal/signal.go:152 +0x36
        goroutine 135962 [select, "job":"CHANGEFEED id=1122401964754698241", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc005b936c0, {0x894b5078, 0xc00bd067d0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 135959
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        goroutine 135963 [select, "job":"CHANGEFEED id=1122401964754698241", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc005b93720, {0x894b5078, 0xc00bd06820})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 135959
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194: 
        Leaked goroutine: goroutine 135962 [select, "job":"CHANGEFEED id=1122401964754698241", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc005b936c0, {0x894b5078, 0xc00bd067d0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 135959
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        Leaked goroutine: goroutine 135963 [select, "job":"CHANGEFEED id=1122401964754698241", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc005b93720, {0x894b5078, 0xc00bd06820})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 135959
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (71.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20715674?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20715674?buildTab=artifacts#/) on master @ [ae0b3d742f343186aa770c4ba8c3bf748bbc78f9](https://github.com/cockroachdb/cockroach/commits/ae0b3d742f343186aa770c4ba8c3bf748bbc78f9):


```
        main.main()
        	bazel-out/s390x-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:804 +0x5b4
        goroutine 23 [select]:
        go.opencensus.io/stats/view.(*worker).start(0xc00112d580)
        	external/io_opencensus_go/stats/view/worker.go:292 +0xba
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0xd0
        goroutine 51 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x82
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x30
        goroutine 52 [chan receive, 6 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0xa8
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x42
        goroutine 27 [syscall, 6 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x3e
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x24
        created by os/signal.Notify.func1.1 in goroutine 52
        	GOROOT/src/os/signal/signal.go:152 +0x36
        goroutine 28 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0xe0
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x30
        goroutine 277625 [select, "job":"CHANGEFEED id=1123017801488203777", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc007ad09c0, {0x894b5b78, 0xc00d201c70})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 277622
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        goroutine 277626 [select, "job":"CHANGEFEED id=1123017801488203777", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc007ad0a40, {0x894b5b78, 0xc00d201cc0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 277622
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194: 
        Leaked goroutine: goroutine 277625 [select, "job":"CHANGEFEED id=1123017801488203777", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc007ad09c0, {0x894b5b78, 0xc00d201c70})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 277622
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        Leaked goroutine: goroutine 277626 [select, "job":"CHANGEFEED id=1123017801488203777", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc007ad0a40, {0x894b5b78, 0xc00d201cc0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 277622
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (238.54s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20738553?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20738553?buildTab=artifacts#/) on master @ [930e2ef1de9b1220f52b6d15663b9e067778c76f](https://github.com/cockroachdb/cockroach/commits/930e2ef1de9b1220f52b6d15663b9e067778c76f):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:806 +0x384
        goroutine 66 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x4001175600)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 52 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 53 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 82 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 53
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 54 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 196655 [select, "job":"CHANGEFEED id=1124242687812534273", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4006b03660, {0x7d080f8, 0x40017226e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 196652
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 196656 [select, "job":"CHANGEFEED id=1124242687812534273", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4006b036c0, {0x7d080f8, 0x4001722730})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 196652
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 196655 [select, "job":"CHANGEFEED id=1124242687812534273", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4006b03660, {0x7d080f8, 0x40017226e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 196652
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 196656 [select, "job":"CHANGEFEED id=1124242687812534273", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4006b036c0, {0x7d080f8, 0x4001722730})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 196652
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (56.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20775135?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20775135?buildTab=artifacts#/) on master @ [9425e71bbcf7cfad3a258dd980c7224b85558ede](https://github.com/cockroachdb/cockroach/commits/9425e71bbcf7cfad3a258dd980c7224b85558ede):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:810 +0x384
        goroutine 50 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x400118b780)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 9 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 10 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 82 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 10
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 11 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 88634 [select, "job":"CHANGEFEED id=1127116309147746305", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4001d86be0, {0x7d1ebb8, 0x400a2f2000})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 88630
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 88633 [select, "job":"CHANGEFEED id=1127116309147746305", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4001d86b40, {0x7d1ebb8, 0x40013b3ef0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 88630
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 88633 [select, "job":"CHANGEFEED id=1127116309147746305", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4001d86b40, {0x7d1ebb8, 0x40013b3ef0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 88630
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 88634 [select, "job":"CHANGEFEED id=1127116309147746305", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4001d86be0, {0x7d1ebb8, 0x400a2f2000})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 88630
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (53.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20801651?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20801651?buildTab=artifacts#/) on master @ [ad61ece183b048d88231301b315c6f4e48afa3cf](https://github.com/cockroachdb/cockroach/commits/ad61ece183b048d88231301b315c6f4e48afa3cf):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:810 +0x384
        goroutine 69 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x40011a5780)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 56 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 57 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 45 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 57
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 58 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 193919 [select, "job":"CHANGEFEED id=1129350006577135617", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400c8c6aa0, {0x7d48518, 0x400a7e6640})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 193915
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 193918 [select, "job":"CHANGEFEED id=1129350006577135617", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400c8c6a40, {0x7d48518, 0x400a7e65f0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 193915
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 193918 [select, "job":"CHANGEFEED id=1129350006577135617", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400c8c6a40, {0x7d48518, 0x400a7e65f0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 193915
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 193919 [select, "job":"CHANGEFEED id=1129350006577135617", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400c8c6aa0, {0x7d48518, 0x400a7e6640})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 193915
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (55.90s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20805692?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20805692?buildTab=artifacts#/) on master @ [8f079ba3f73e5b7f4e3573f9a9d7eb62080b6f81](https://github.com/cockroachdb/cockroach/commits/8f079ba3f73e5b7f4e3573f9a9d7eb62080b6f81):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:810 +0x384
        goroutine 65 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x400135d800)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 69 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 70 [chan receive, 1 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 82 [syscall, 1 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 70
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 21 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 131363 [select, "job":"CHANGEFEED id=1129555746049163265", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4004a5bbe0, {0x7cf6f58, 0x400dafc8c0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 131312
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 131364 [select, "job":"CHANGEFEED id=1129555746049163265", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4004a5bc40, {0x7cf6f58, 0x400dafc910})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 131312
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 131363 [select, "job":"CHANGEFEED id=1129555746049163265", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4004a5bbe0, {0x7cf6f58, 0x400dafc8c0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 131312
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 131364 [select, "job":"CHANGEFEED id=1129555746049163265", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4004a5bc40, {0x7cf6f58, 0x400dafc910})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 131312
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (46.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20856992?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20856992?buildTab=artifacts#/) on master @ [0f9fc01394eb3d00db06ff0587b00d05375822fe](https://github.com/cockroachdb/cockroach/commits/0f9fc01394eb3d00db06ff0587b00d05375822fe):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:836 +0x384
        goroutine 66 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x40012c9400)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 51 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 52 [chan receive, 1 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 68 [syscall, 1 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 52
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 53 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 129095 [select, "job":"CHANGEFEED id=1132121517030539265", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007247aa0, {0x7d54cb8, 0x400225edc0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 129092
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 129096 [select, "job":"CHANGEFEED id=1132121517030539265", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007247b00, {0x7d54cb8, 0x400225ee10})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 129092
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 129095 [select, "job":"CHANGEFEED id=1132121517030539265", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007247aa0, {0x7d54cb8, 0x400225edc0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 129092
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 129096 [select, "job":"CHANGEFEED id=1132121517030539265", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007247b00, {0x7d54cb8, 0x400225ee10})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 129092
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (47.50s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #159177 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot T-cdc branch-release-25.4.2-rc release-blocker]
- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20875528?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20875528?buildTab=artifacts#/) on master @ [48e75432cb500245f42eac0e574ed4004be96bd7](https://github.com/cockroachdb/cockroach/commits/48e75432cb500245f42eac0e574ed4004be96bd7):


```
        main.main()
        	bazel-out/s390x-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:846 +0x5b4
        goroutine 34 [select]:
        go.opencensus.io/stats/view.(*worker).start(0xc0011cf700)
        	external/io_opencensus_go/stats/view/worker.go:292 +0xba
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0xd0
        goroutine 38 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x82
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x30
        goroutine 39 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0xa8
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x42
        goroutine 51 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x3e
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x24
        created by os/signal.Notify.func1.1 in goroutine 39
        	GOROOT/src/os/signal/signal.go:152 +0x36
        goroutine 40 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0xe0
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x30
        goroutine 199870 [select, "job":"CHANGEFEED id=1133578747084963841", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0025676e0, {0x895aa7b8, 0xc008ef5090})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 199867
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        goroutine 199871 [select, "job":"CHANGEFEED id=1133578747084963841", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0025677a0, {0x895aa7b8, 0xc008ef50e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 199867
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194: 
        Leaked goroutine: goroutine 199870 [select, "job":"CHANGEFEED id=1133578747084963841", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0025676e0, {0x895aa7b8, 0xc008ef5090})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 199867
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        Leaked goroutine: goroutine 199871 [select, "job":"CHANGEFEED id=1133578747084963841", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0025677a0, {0x895aa7b8, 0xc008ef50e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 199867
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (74.57s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #159720 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot T-cdc branch-release-26.1 release-blocker]
- #159177 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.4.2-rc]
- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21032925?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21032925?buildTab=artifacts#/) on master @ [f142ecb6d44fe1b8b68fe8c465e2f6e8ffc21507](https://github.com/cockroachdb/cockroach/commits/f142ecb6d44fe1b8b68fe8c465e2f6e8ffc21507):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:852 +0x384
        goroutine 33 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x4001205780)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 53 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 39 [chan receive, 4 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 68 [syscall, 4 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 39
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 40 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 195940 [select, "job":"CHANGEFEED id=1145270063775875073", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4008a63080, {0x7ec0e78, 0x4006346d20})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 195895
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 195941 [select, "job":"CHANGEFEED id=1145270063775875073", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4008a63100, {0x7ec0e78, 0x4006346d70})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 195895
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 195940 [select, "job":"CHANGEFEED id=1145270063775875073", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4008a63080, {0x7ec0e78, 0x4006346d20})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 195895
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 195941 [select, "job":"CHANGEFEED id=1145270063775875073", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4008a63100, {0x7ec0e78, 0x4006346d70})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 195895
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (70.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #159720 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1]
- #159177 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.4.2-rc]
- #156850 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.2.9-rc]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #148151 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.2-rc]
- #147047 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2.1-rc]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21165752?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21165752?buildTab=artifacts#/) on master @ [c917093a0387cf08bec3cc8407ee2913200d3255](https://github.com/cockroachdb/cockroach/commits/c917093a0387cf08bec3cc8407ee2913200d3255):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:860 +0x384
        goroutine 13 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x40011eee80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 29 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 30 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 66 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 30
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 31 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 91718 [select, "job":"CHANGEFEED id=1153661399427219457", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007490280, {0x7a5f518, 0x400919c690})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 91715
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 91719 [select, "job":"CHANGEFEED id=1153661399427219457", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007490300, {0x7a5f518, 0x400919c6e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 91715
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 91718 [select, "job":"CHANGEFEED id=1153661399427219457", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007490280, {0x7a5f518, 0x400919c690})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 91715
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 91719 [select, "job":"CHANGEFEED id=1153661399427219457", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4007490300, {0x7a5f518, 0x400919c6e0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 91715
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (57.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #163886 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1.1-rc s390x-test-failure]
- #159720 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21191148?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21191148?buildTab=artifacts#/) on master @ [31bd4284d8b9d22128b628702254fd5db11870fa](https://github.com/cockroachdb/cockroach/commits/31bd4284d8b9d22128b628702254fd5db11870fa):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:860 +0x384
        goroutine 40 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x400114ee80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 44 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 45 [chan receive, 3 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 51 [syscall, 3 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 45
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 46 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 146720 [select, 1 minutes, "job":"CHANGEFEED id=1155135165182935041", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400d880900, {0x7a735f8, 0x4008ebeb90})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 146716
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 146719 [select, 1 minutes, "job":"CHANGEFEED id=1155135165182935041", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400d8808a0, {0x7a735f8, 0x4008ebeb40})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 146716
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 146719 [select, 1 minutes, "job":"CHANGEFEED id=1155135165182935041", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400d8808a0, {0x7a735f8, 0x4008ebeb40})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 146716
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 146720 [select, 1 minutes, "job":"CHANGEFEED id=1155135165182935041", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x400d880900, {0x7a735f8, 0x4008ebeb90})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 146716
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (116.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #163886 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1.1-rc s390x-test-failure]
- #159720 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21206159?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21206159?buildTab=artifacts#/) on master @ [0f1fac7e7240aff771e0cb4f1a259245c7733eaf](https://github.com/cockroachdb/cockroach/commits/0f1fac7e7240aff771e0cb4f1a259245c7733eaf):


```
        main.main()
        	bazel-out/aarch64-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:864 +0x384
        goroutine 40 [select]:
        go.opencensus.io/stats/view.(*worker).start(0x40011ece80)
        	external/io_opencensus_go/stats/view/worker.go:292 +0x84
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0x94
        goroutine 44 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x50
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x24
        goroutine 45 [chan receive, 2 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0x70
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x30
        goroutine 66 [syscall, 2 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x30
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x1c
        created by os/signal.Notify.func1.1 in goroutine 45
        	GOROOT/src/os/signal/signal.go:152 +0x28
        goroutine 46 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0x8c
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x24
        goroutine 206029 [select, "job":"CHANGEFEED id=1155715266348810241", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40152afd40, {0x7a92558, 0x40036b6780})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 206025
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        goroutine 206028 [select, "job":"CHANGEFEED id=1155715266348810241", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40152afce0, {0x7a92558, 0x40036b6730})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 206025
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c: 
        Leaked goroutine: goroutine 206028 [select, "job":"CHANGEFEED id=1155715266348810241", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40152afce0, {0x7a92558, 0x40036b6730})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 206025
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
        Leaked goroutine: goroutine 206029 [select, "job":"CHANGEFEED id=1155715266348810241", "n":"sql1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x40152afd40, {0x7a92558, 0x40036b6780})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xbc
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 206025
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x10c
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (86.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #163886 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1.1-rc s390x-test-failure]
- #159720 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedEnrichedSourceWithDataJSON [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21233309?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21233309?buildTab=artifacts#/) on master @ [941dc3774136f185ea54978e4c8cf16dfe05c63e](https://github.com/cockroachdb/cockroach/commits/941dc3774136f185ea54978e4c8cf16dfe05c63e):


```
        main.main()
        	bazel-out/s390x-fastbuild/bin/pkg/ccl/changefeedccl/changefeedccl_test_/testmain.go:866 +0x5b4
        goroutine 10 [select]:
        go.opencensus.io/stats/view.(*worker).start(0xc001156e00)
        	external/io_opencensus_go/stats/view/worker.go:292 +0xba
        created by go.opencensus.io/stats/view.init.0 in goroutine 1
        	external/io_opencensus_go/stats/view/worker.go:34 +0xd0
        goroutine 19 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
        	pkg/util/log/log_flush.go:123 +0x82
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:78 +0x30
        goroutine 20 [chan receive, 1 minutes]:
        github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
        	pkg/util/log/log_flush.go:147 +0xa8
        created by github.com/cockroachdb/cockroach/pkg/util/log.init.5 in goroutine 1
        	pkg/util/log/log_flush.go:79 +0x42
        goroutine 14 [syscall, 1 minutes]:
        os/signal.signal_recv()
        	GOROOT/src/runtime/sigqueue.go:152 +0x3e
        os/signal.loop()
        	GOROOT/src/os/signal/signal_unix.go:23 +0x24
        created by os/signal.Notify.func1.1 in goroutine 20
        	GOROOT/src/os/signal/signal.go:152 +0x36
        goroutine 15 [chan receive]:
        github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0.func1()
        	pkg/util/goschedstats/runnable_enabled.go:144 +0xe0
        created by github.com/cockroachdb/cockroach/pkg/util/goschedstats.init.0 in goroutine 1
        	pkg/util/goschedstats/runnable_enabled.go:136 +0x30
        goroutine 139725 [select, "job":"CHANGEFEED id=1157480727977820161", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0032bc120, {0x8921c078, 0xc01826d450})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 139722
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        goroutine 139726 [select, "job":"CHANGEFEED id=1157480727977820161", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0032bc180, {0x8921c078, 0xc01826d4a0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 139722
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194: 
        Leaked goroutine: goroutine 139725 [select, "job":"CHANGEFEED id=1157480727977820161", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0032bc120, {0x8921c078, 0xc01826d450})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 139722
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
        Leaked goroutine: goroutine 139726 [select, "job":"CHANGEFEED id=1157480727977820161", "n":"1"]:
        google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0xc0032bc180, {0x8921c078, 0xc01826d4a0})
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0x104
        created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 139722
        	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x194
--- FAIL: TestChangefeedEnrichedSourceWithDataJSON (53.48s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #163886 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1.1-rc s390x-test-failure]
- #159720 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-26.1]
- #155548 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.4]
- #152378 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-25.3]
- #144587 ccl/changefeedccl: TestChangefeedEnrichedSourceWithDataJSON failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-25.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedEnrichedSourceWithDataJSON.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_processors.go
+++ b/pkg/ccl/changefeedccl/changefeed_processors.go
@@ -426,7 +426,13 @@ func (ca *changeAggregator) Start(ctx context.Context) {
 	}
 
 	ca.sink = &errorWrapperSink{wrapped: ca.sink}
-	ca.eventConsumer, ca.sink, err = newEventConsumer(
+
+	// Use local variables so that ca.sink is not overwritten with nil on error.
+	// newEventConsumer returns (nil, nil, err) on failure, and overwriting
+	// ca.sink would prevent close() from cleaning up the already-dialed sink.
+	var consumer eventConsumer
+	var s EventSink
+	consumer, s, err = newEventConsumer(
 		ctx, ca.FlowCtx.Cfg, ca.spec, feed, ca.frontier, kvFeedHighWater,
 		ca.sink, ca.metrics, ca.sliMetrics, ca.knobs)
 	if err != nil {
@@ -435,6 +441,10 @@ func (ca *changeAggregator) Start(ctx context.Context) {
 		ca.cancel()
 		return
 	}
+	ca.eventConsumer = consumer
+	// Reassign sink, since s may be a wrapped version of the original sink.
+	ca.sink = s
+
 	// Generate expensive checkpoint only after we ran for a while.
 	ca.lastSpanFlush = timeutil.Now()
 }

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -13623,3 +13623,122 @@ func TestChangefeedServerlessLocalityFilter(t *testing.T) {
 		`foo: [1]->{"after": {"pk": 1}}`,
 	})
 }
+
+// TestSinkClosedOnEventConsumerError verifies that when newEventConsumer fails
+// during changeAggregator.Start(), the sink is still properly closed. This is a
+// regression test for #144102 where newEventConsumer's error path returned
+// (nil, nil, err), overwriting ca.sink with nil and preventing cleanup.
+func TestSinkClosedOnEventConsumerError(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	s, stopServer := makeServer(t,
+		withAllowChangefeedErr("injected event consumer error expected"))
+	defer stopServer()
+
+	sqlDB := sqlutils.MakeSQLRunner(s.DB)
+	knobs := s.TestingKnobs.
+		DistSQL.(*execinfra.TestingKnobs).
+		Changefeed.(*TestingKnobs)
+
+	// Track how many times Dial and Close are called across retries.
+	var dialCount, closeCount atomic.Int32
+	knobs.WrapSink = func(s Sink, _ jobspb.JobID) Sink {
+		return &closeTrackingSink{
+			wrapped:    s,
+			dialCount:  &dialCount,
+			closeCount: &closeCount,
+		}
+	}
+
+	// Only inject errors a fixed number of times so the changefeed eventually
+	// stabilises and the counters stop moving.
+	const numErrors = 3
+	var errorCount atomic.Int32
+	knobs.EventConsumerError = func() error {
+		if errorCount.Add(1) <= numErrors {
+			return errors.New("injected event consumer error")
+		}
+		return nil
+	}
+
+	sqlDB.Exec(t, `CREATE TABLE foo (pk INT PRIMARY KEY)`)
+	sqlDB.Exec(t, `INSERT INTO foo VALUES (1)`)
+
+	// The changefeed will retry numErrors times, then succeed. Wait for all
+	// injected errors to fire before cancelling.
+	sqlDB.Exec(t, `CREATE CHANGEFEED FOR foo INTO 'null://'`)
+
+	testutils.SucceedsSoon(t, func() error {
+		if errorCount.Load() < numErrors {
+			return errors.Newf("waiting for all injected errors (errorCount=%d)", errorCount.Load())
+		}
+		return nil
+	})
+
+	// Verify the minimum expected dials occurred: 1 (validateSink) +
+	// numErrors (failed) + 1 (successful). Under stress there may be more
+	// due to other retriable errors (e.g. enriched envelope descriptor lookups).
+	require.GreaterOrEqual(t, dialCount.Load(), int32(numErrors+2))
+
+	// Cancel the changefeed so all sinks are closed and counters stabilize.
+	var jobID int64
+	sqlDB.QueryRow(t,
+		`SELECT job_id FROM [SHOW CHANGEFEED JOBS] ORDER BY created DESC LIMIT 1`,
+	).Scan(&jobID)
+	sqlDB.Exec(t, `CANCEL JOB $1`, jobID)
+
+	// After cancellation, every dialed sink must have been closed.
+	testutils.SucceedsSoon(t, func() error {
+		d := dialCount.Load()
+		c := closeCount.Load()
+		if c != d {
+			return errors.Newf("sink dialed %d times but closed only %d times", d, c)
+		}
+		return nil
+	})
+}
+
+// closeTrackingSink wraps a Sink and counts Dial and Close calls.
+type closeTrackingSink struct {
+	wrapped    Sink
+	dialCount  *atomic.Int32
+	closeCount *atomic.Int32
+}
+
+var _ Sink = (*closeTrackingSink)(nil)
+
+func (s *closeTrackingSink) Dial() error {
+	s.dialCount.Add(1)
+	return s.wrapped.Dial()
+}
+
+func (s *closeTrackingSink) Close() error {
+	s.closeCount.Add(1)
+	return s.wrapped.Close()
+}
+
+func (s *closeTrackingSink) EmitRow(
+	ctx context.Context,
+	topic TopicDescriptor,
+	key, value []byte,
+	updated, mvcc hlc.Timestamp,
+	alloc kvevent.Alloc,
+	headers rowHeaders,
+) error {
+	return s.wrapped.EmitRow(ctx, topic, key, value, updated, mvcc, alloc, headers)
+}
+
+func (s *closeTrackingSink) Flush(ctx context.Context) error {
+	return s.wrapped.Flush(ctx)
+}
+
+func (s *closeTrackingSink) EmitResolvedTimestamp(
+	ctx context.Context, encoder Encoder, resolved hlc.Timestamp,
+) error {
+	return s.wrapped.EmitResolvedTimestamp(ctx, encoder, resolved)
+}
+
+func (s *closeTrackingSink) getConcreteType() sinkType {
+	return s.wrapped.getConcreteType()
+}

--- a/pkg/ccl/changefeedccl/event_processing.go
+++ b/pkg/ccl/changefeedccl/event_processing.go
@@ -96,6 +96,11 @@ func newEventConsumer(
 	sliMetrics *sliMetrics,
 	knobs TestingKnobs,
 ) (eventConsumer, EventSink, error) {
+	if knobs.EventConsumerError != nil {
+		if err := knobs.EventConsumerError(); err != nil {
+			return nil, nil, err
+		}
+	}
 	encodingOpts, err := feed.Opts.GetEncodingOptions()
 	if err != nil {
 		return nil, nil, err

--- a/pkg/ccl/changefeedccl/helpers_test.go
+++ b/pkg/ccl/changefeedccl/helpers_test.go
@@ -1378,8 +1378,7 @@ func maybeForceEnrichedEnvelope(
 	// - sinkless feeds can't be tracked by job id
 	// - sql & pulsar are not supported
 	// - cloudstorage uses parquet sometimes which complicates things
-	// - pubsub feeds have an issue with leaking goroutines; see #144102
-	case *sinklessFeedFactory, *tableFeedFactory, *pulsarFeedFactory, *cloudFeedFactory, *pubsubFeedFactory:
+	case *sinklessFeedFactory, *tableFeedFactory, *pulsarFeedFactory, *cloudFeedFactory:
 		t.Logf("did not force enriched envelope for %s because %T is not supported", create, f)
 		return create, args, false, nil
 	}

--- a/pkg/ccl/changefeedccl/testing_knobs.go
+++ b/pkg/ccl/changefeedccl/testing_knobs.go
@@ -120,6 +120,11 @@ type TestingKnobs struct {
 	// OverrideCursorAge is used to change how old a cursor is. Returns time in nanoseconds.
 	OverrideCursorAge func() int64
 
+	// EventConsumerError, if set, is called before creating the event consumer.
+	// If it returns a non-nil error, newEventConsumer returns that error.
+	// This is useful for testing error handling in changeAggregator.Start().
+	EventConsumerError func() error
+
 	// MakeKVFeedToAggregatorBufferKnobs is used to make a fresh set of testing knobs
 	// to pass to the constructor of the kv feed to change aggregator buffer.
 	MakeKVFeedToAggregatorBufferKnobs func() kvevent.BlockingBufferTestingKnobs

